File: src/RehabEasy.Domain/Models/PatientTestHistoryEntry.cs
namespace RehabEasy.Domain.Models;

public sealed class PatientTestHistoryEntry
{
    public string RecordId { get; init; } = string.Empty;
    public string TestType { get; init; } = string.Empty;
    public string Title { get; init; } = string.Empty;
    public DateTimeOffset ReceivedAt { get; init; }
    public string MetricsSummary { get; init; } = string.Empty;
    public string DetailText { get; init; } = string.Empty;
}

File: java/src/main/java/com/rehabeasy/model/PatientTestHistoryEntry.java
package com.rehabeasy.model;

import java.time.Instant;

public record PatientTestHistoryEntry(
        String recordId,
        String testType,
        String title,
        Instant receivedAt,
        String metricsSummary,
        String detailText
) {
    public PatientTestHistoryEntry {
        recordId = recordId == null ? "" : recordId;
        testType = testType == null ? "" : testType;
        title = title == null ? "" : title;
        receivedAt = receivedAt == null ? Instant.EPOCH : receivedAt;
        metricsSummary = metricsSummary == null ? "" : metricsSummary;
        detailText = detailText == null ? "" : detailText;
    }
}
